### DISCUSSION TITLE
OpenSumi 文件服务如何对接阿里云 oss

### DISCUSSION THREAD

**Discussion by songxc** [opensumi/core#2064] [Q&A]
我基于官方示例（纯前端版本）在 file-provider/ 中改写了 http-file.service.ts，核心覆写了 readDir、readFile、updateFile、createFile、deleteFile 等几个 API，通过 oss sdk 打通了 oss 存储，发现可以正常创建、修改、删除文件吗。但是遇到了几个问题：
1. 问题一：支持单文件、工程两种模式，但是目前没有找到如何实现单文件模式（只允许修改指定文件内容，不允许创建、删除）；
2. 问题二：OpenSumi 会默认加载浏览器文件，不会通过 http 读取文件，希望默认走网络；

**Comment by erha19** [comment]
@songxc 

问题一：目前框架侧不会区分业务属性时单文件还是工程，如果希望某个文件仅只读，可以参考自定义只读模式文档 https://opensumi.com/zh/docs/integrate/universal-integrate-case/custom-readonly-mode 自行实现，如果是基于上面讲的 `只允许修改指定文件内容，不允许创建、删除` ，则需要你自行在 `http-file.service.ts` 服务中处理，不处理 createFile、deleteFile 就行了

问题二：不太了解具体意思，默认加载的浏览器文件是指哪些？

**Reply by songxc** [reply]
问题一：我目前的做法确实是在 createFile、deleteFile 中做了判断和拦截，但是发现创建文件夹的话是不会进入到 createFile、deleteFile 的，拦不到。更理想的做法是可以控制「创建文件」和「创建文件夹」的入口，但没有找到哪里可以定制。
<img width="61" alt="image" src="https://user-images.githubusercontent.com/4737493/206116802-f494518f-54ff-4c8f-9730-f447195f5603.png">

问题二：大概意思是我要打开的工程是个远程的工程，所有的文件读写都是直接操作远程的，不希望有 browserFS 这一层。

**Reply by erha19** [reply]
1. 这种情况可以参考 https://opensumi.com/zh/docs/integrate/universal-integrate-case/custom-readonly-mode ，把插件文件的命令卸载掉，这样界面上就不会有响应的功能按钮，如 `FILE_COMMANDS.NEW_FILE` 和 `FILE_COMMANDS.NEW_FOLDER`
2. browserFS 主要用于在浏览器内存上模拟 Node 层的 FS 文件系统，示例中的文件读写仍然是通过网络请求的的，目前不支持移除这层
